src/Microsoft.OData.Core/Json/ODataJsonReaderCoreUtils.cs

```diff
@@ -121,6 +122,109 @@ internal static class ODataJsonReaderCoreUtils
             return false;
         }
 
+        /// <summary>
+        /// Try and parse spatial type from the json payload.
+        /// </summary>
+        /// <param name="jsonReader">The JSON reader to read from.</param>
+        /// <param name="insideJsonObjectValue">true if the reader is positioned on the first property of the value which is a JSON Object
+        ///     (or the second property if the first one was odata.type).</param>
+        /// <param name="inputContext">The input context with all the settings.</param>
+        /// <param name="expectedValueTypeReference">Expected edm property type.</param>
+        /// <param name="validateNullValue">true to validate null values; otherwise false.</param>
+        /// <param name="recursionDepth">The recursion depth to start with.</param>
+        /// <param name="propertyName">The name of the property whose value is being read, if applicable (used for error reporting).</param>
+        /// <returns>An instance of the spatial type.</returns>
+        internal static async Task<ISpatial> ReadSpatialValueAsync(
+            IJsonReaderAsync jsonReader,
+            bool insideJsonObjectValue,
+            ODataInputContext inputContext,
+            IEdmPrimitiveTypeReference expectedValueTypeReference,
+            bool validateNullValue,
+            int recursionDepth,
+            string propertyName)
+        {
+            Debug.Assert(jsonReader != null, "jsonReader != null");
+            Debug.Assert(inputContext != null, "inputContext != null");
+            Debug.Assert(expectedValueTypeReference != null, "expectedValueTypeReference != null");
+            Debug.Assert(expectedValueTypeReference.IsSpatial(), "TryParseSpatialType must be called only with spatial types");
+
+            // Spatial value can be either null constant or a JSON object
+            // If it's a null primitive value, report a null value.
+            if (!insideJsonObjectValue
+                && await TryReadNullValueAsync(jsonReader, inputContext, expectedValueTypeReference, validateNullValue, propertyName)
+                .ConfigureAwait(false))
+            {
+                return null;
+            }
+
+            ISpatial spatialValue = null;
+            if (insideJsonObjectValue || jsonReader.NodeType == JsonNodeType.StartObject)
+            {
+                IDictionary<string, object> jsonObject = await ReadObjectValueAsync(jsonReader, insideJsonObjectValue, inputContext, recursionDepth)

```

Comment:
```suggestion
                Dictionary<string, object> jsonObject = await ReadObjectValueAsync(jsonReader, insideJsonObjectValue, inputContext, recursionDepth)

```

Reply:
@odero This won't work because the `ReadObjectValueAsync` method returns `IDictionary<string, object>` and that can't be implicitly converted to `Dictionary<string, object>`.

Reply:
Since the method is private can't we make the change without any issues? IDictionary may lead to a bit of boxing IIRC.

Reply:
@marabooy Point